deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/scala/org/deeplearning4j/spark/ml/impl/AutoEncoderWrapper.scala

```diff
@@ -48,7 +48,7 @@ abstract class AutoEncoderWrapper[E <: AutoEncoderWrapper[E, M], M <: AutoEncode
         val set2 = dataset.select($(inputCol)).rdd.map(mapVectorFunc)
         val ds = set2.map(v => new DataSet(Nd4j.create(v.toArray), Nd4j.create(v.toArray)))
         val fitted = sparkNet.fit(ds)
-        sparkNet
+        fitted.clone()
     }
```

Comment:
Doesn't this get expensive? The averaged network isn't used anywhere else. Do we really need to clone?